src/main/java/uk/gov/companieshouse/insolvency/data/converter/RoleEnumReadingConverter.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.companieshouse.insolvency.data.converter;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.convert.ReadingConverter;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.insolvency.CaseDates;
+import uk.gov.companieshouse.api.insolvency.Practitioners;
+
+@Component
+@ReadingConverter
+public class RoleEnumReadingConverter implements Converter<String, Practitioners.RoleEnum> {
```

Comment:
I think we can have a single generic Enum reading/writing converter for all enum types, please check the PoC branch I created